frame/nomination-pools/src/lib.rs

```diff
@@ -538,19 +538,104 @@ pub struct PoolRoles<AccountId> {
 	pub state_toggler: Option<AccountId>,
 }
 
+/// Pool commission.
+///
+/// The pool depositor can set a commission upon a pool creation, from which the pool owner can
+/// update thereafter. The `max` commission value can only be decreased after the initial value is
+/// set, as to prevent the commission from repeatedly increasing.
+///
+/// A commission throttle is also optional, allowing the pool to set strict limits to how much
+/// commission can change in each update, and how often updates can take place.
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, DebugNoBound, PartialEq, Copy, Clone)]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct Commission<T: Config> {
+	/// The active commission rate of the pool along with the account commission is paid to.
+	current: Option<(Perbill, T::AccountId)>,
+	/// An optional maximum commission that can be set by the pool root. Once set, this value
+	/// cannot be updated.
+	max: Option<Perbill>,
+	/// Configiration around how often the commission can be updated, and metadata around the
+	/// previous round of updates.
+	throttle: Option<CommissionThrottle<T>>,
+}
+
+impl<T: Config> Default for Commission<T> {
+	fn default() -> Self {
+		Self { current: None, max: None, throttle: None }
+	}
+}
+
+/// Pool commission throttle preferences.
+///
+/// Throttle prefs need to be passed and configured together. This struct is used in
+/// the `set_commission_throttle` call as well as in CommissionThrottle.
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Debug, PartialEq, Copy, Clone)]
+pub struct CommissionThrottlePrefs<BlockNumber> {
+	/// The maximum amount the commission can be updated by.
+	pub max_increase: Perbill,
+	/// How often an update can take place.
+	pub min_delay: BlockNumber,
+}
+
+/// The pool root is able to set a commission throttle for their pool.
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, DebugNoBound, PartialEq, Copy, Clone)]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct CommissionThrottle<T: Config> {
+	///  The change rate dictates how often and by how much commission can be updated.
+	pub change_rate: CommissionThrottlePrefs<T::BlockNumber>,
+	/// The block the previous commission update took place.
+	previous_set_at: Option<T::BlockNumber>,
+}
+/// The commission throttle defaults, used for readability and convenience.
+/// `change_rate` is not utilised in this code, but the provided values would
+/// be quite sensible for throttling.
+impl<T: Config> Default for CommissionThrottle<T> {
+	fn default() -> Self {
+		Self {
+			change_rate: CommissionThrottlePrefs {
+				max_increase: Perbill::from_percent(1),
```

Comment:
the default IMO should be "maximum change possible" i.e. "No limit at all".

Reply:
Actually, this type should NOT have a default IMO, because it is hard to justify why some random value is the default.